crowd_anki/export/anki_exporter.py

```diff
@@ -67,3 +71,81 @@ def _copy_media(self, deck, deck_directory):
+            return notes
+
+        sort_method = deck_sort_config["method"]
+        is_reversed = deck_sort_config["reversed"] if "reversed" in deck_sort_config else False
+
+        def _formatted_sort_method(method):
+            return re.sub("[-_\s+]", "", method).lower()
+
+        # Make sure formatting of sort_method is correct, and format each entry
+        if isinstance(sort_method, str):
+            sort_method = [_formatted_sort_method(sort_method)]
+        elif isinstance(sort_method, list):
+            if len(sort_method) == 1:
+                sort_method = [_formatted_sort_method(sort_method[0])]
+            elif len(sort_method) > 1:
+                sort_method = [_formatted_sort_method(sm) for sm in sort_method]                
+            else:
+                return notes
+        else:
+            return notes
+
+        # Create mapping of sortable_values variable, to sort_method keys, to sortable_values lambda
+        sorting_definitions = [
+            (0, ("guid", "guids"), lambda i: i.anki_object.guid),
+            (1, ("flag", "flags"), lambda i: i.anki_object.flags),
+            (2, ("tag", "tags"), lambda i: i.anki_object.tags),
+            (3, ("notemodel", "notemodels", "notemodelname", "notemodelsname", "notemodelnames", "notemodelsnames"), lambda i: i.anki_object._model["name"]),
+            (4, ("notemodelid", "notemodelsid", "notemodelids", "notemodelsids"), lambda i: i.anki_object._model["crowdanki_uuid"]),
+            (5, ("field1", "firstfield", "first"), lambda i: i.anki_object.fields[0]),
+            (6, ("field2", "secondfield", "second"), lambda i: i.anki_object.fields[1])
+        ]
+
+        # Extract each notes values which could be used for sorting into sortable_values, for use later on
+        for note in notes:
+            note.sortable_values = [None] * len(sorting_definitions)
+            note.export_filter_set.add("sortable_values")
+
+            for defs in sorting_definitions:
+                note.sortable_values[defs[0]] = defs[2](note)
+
+
+        # Flatten out the sorting_definitions to a map with one value per entry
+        sort_map = {singlekey:num for num, keytuple, lam in sorting_definitions for singlekey in keytuple}
+
+
+        if sort_method[0] in ["default", "none", "nosorting"]:      # Only the first method is considered for these pass variables
+            if not is_reversed:
+                pass
+            else:
+                notes = list(reversed(notes))
+        else:
+            all_methods_good = True
+            for method in sort_method:
+                if method not in sort_map:
+                    all_methods_good = False
+                    break
+            
+            # Only continue if all sorting methods are known
+            if all_methods_good:
+                sort_numbers = [sort_map[sm] for sm in sort_method]
+
+                if len(sort_numbers) == 1:
```

Comment:
why can't this be a loop/comprehension?

Reply:
I wanted it to be so, but could not see how to create a tuple of various length based on the length of sort_numbers. Is that possible?

Reply:
Use tuple comprehension? Or create a list and convert it to tuple